client/workflow.go

```diff
@@ -0,0 +1,272 @@
+}
+
+type StartWorkflowResponse struct {
+	InstanceID string
+}
+
+type GetWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type GetWorkflowResponse struct {
+	InstanceID    string
+	WorkflowName  string
+	CreatedAt     time.Time
+	LastUpdatedAt time.Time
+	RuntimeStatus string
+	Properties    map[string]string
+}
+
+type PurgeWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type TerminateWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type PauseWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type ResumeWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+}
+
+type RaiseEventWorkflowRequest struct {
+	InstanceID        string
+	WorkflowComponent string
+	EventName         string
+	EventData         any
+	SendRawData       bool // Set to True in order to disable serialization on the data
+}
+
+// StartWorkflowBeta1 starts a workflow using the beta1 spec.
+func (c *GRPCClient) StartWorkflowBeta1(ctx context.Context, req *StartWorkflowRequest) (*StartWorkflowResponse, error) {
+	if req.InstanceID == "" {
+		req.InstanceID = uuid.New().String()
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	if req.WorkflowName == "" {
+		return nil, errors.New("failed to start workflow: WorkflowName must be supplied")
+	}
+
+	var input []byte
+	var err error
+	if req.SendRawInput {
+		input = req.Input.([]byte)
+	} else {
+		input, err = marshalInput(req.Input)
+		if err != nil {
+			return nil, fmt.Errorf("failed to start workflow: %v", err)
+		}
+	}
+
+	resp, err := c.protoClient.StartWorkflowBeta1(c.withAuthToken(ctx), &pb.StartWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+		WorkflowName:      req.WorkflowName,
+		Options:           req.Options,
+		Input:             input,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to start workflow instance: %v", err)
+	}
+	return &StartWorkflowResponse{
+		InstanceID: resp.GetInstanceId(),
+	}, nil
+}
+
+// GetWorkflowBeta1 gets the status of a workflow using the beta1 spec.
+func (c *GRPCClient) GetWorkflowBeta1(ctx context.Context, req *GetWorkflowRequest) (*GetWorkflowResponse, error) {
+	if req.InstanceID == "" {
+		return nil, errors.New("failed to get workflow status: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	resp, err := c.protoClient.GetWorkflowBeta1(c.withAuthToken(ctx), &pb.GetWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get workflow status: %v", err)
+	}
+	if resp.GetCreatedAt() == nil {
+		resp.CreatedAt = timestamppb.Now()
+	}
+	if resp.GetLastUpdatedAt() == nil {
+		resp.LastUpdatedAt = timestamppb.Now()
```

Comment:
I wonder if it makes more sense to leave these timestamp values as nil or as some zero value rather than setting them to `timestamppb.Now()`, which could be confusing.

Reply:
I'll leave them as nil